### PR TITLE
Edit profile

### PR DESCRIPTION
**Pull request by ravikumar0403**
### change log

- users can now edit and update profile details, profile picture and cover image

### CODE REVIEW

File: src/components/Modal.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  return reactDom.createPortal(
+    <div
+      className="absolute top-0 left-0 h-screen w-full backdrop-blur-sm mt-[8vh] flex justify-center z-20"
+      onClick={handleClick}
+    >
+      {type === "post" && <PostInputModal />}
+      {type === "profile" && <EditProfileModal />}
+      {type === "cover" && <EditCoverModal />}
+    </div>,
+    document.body
```
Comment (yashpatel1797) [inline src/components/Modal.js]: good use of createPortal.

File: src/components/BottomNav.js [hunk 2]
```diff
@@ -30,7 +30,11 @@ export const BottomNav = () => {
         >
           <MdExplore size={25} />
         </NavLink>
-        <button to="/" className="px-2" onClick={() => dispatch(showModal())}>
+        <button
+          to="/"
+          className="px-2"
+          onClick={() => dispatch(showModal({ type: "post" }))}
```
Comment (yashpatel1797) [inline src/components/BottomNav.js]: nice way to handle multiple modal. I try to implement this.

File: src/components/EditProfileModal.js [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+    <div className="p-5 mt-12 bg-white h-max shadow rounded border mx-2 max-w-full w-[30rem] dark:bg-slate-800 dark:text-white">
+      <div className="flex justify-between">
+        <h2 className="text-xl ">Edit Profile</h2>
+        <button className="px-3" onClick={() => dispatch(closeModal())}>
+          <IoIosClose size={"1.5rem"} />
+        </button>
+      </div>
+      <Formik
+        initialValues={{ ...user }}
+        onSubmit={handleSubmit}
```
Comment (yashpatel1797) [inline src/components/EditProfileModal.js:40]: like how you implemented `formik` learn something new.

**Review by yashpatel1797 — APPROVED** [review]
code looks clean
file structure is good
design looks consistent

File: src/components/EditCoverModal.js [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+        </button>
+      </div>
+      <div className="my-5 shadow-lg border-2 border-slate-500 rounded flex items-center justify-center">
+        <img
+          src={file ? URL.createObjectURL(file) : user.cover}
+          alt="cover_img"
+        />
+      </div>
+      <div className="flex flex-row-reverse">
+        {!file ? (
```
Comment (pruthvirajmv) [inline src/components/EditCoverModal.js]: here ! check can be eliminated by swapping true & false condition

File: src/components/Modal.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  if (!isVisible) return null;
+
+  return reactDom.createPortal(
+    <div
+      className="absolute top-0 left-0 h-screen w-full backdrop-blur-sm mt-[8vh] flex justify-center z-20"
+      onClick={handleClick}
+    >
+      {type === "post" && <PostInputModal />}
+      {type === "profile" && <EditProfileModal />}
+      {type === "cover" && <EditCoverModal />}
```
Comment (pruthvirajmv) [inline src/components/Modal.js]: suggest you to create constant for modal types

File: src/components/PostInputModal.js [hunk 6]
```diff
@@ -37,62 +36,59 @@ export const PostInputModal = () => {
...
+        ></textarea>
+        <span className="text-right text-xs text-slate-400">
+          {280 - postContent.length} characters remaining
+        </span>
+        <div>
+          {(files.length > 0 || selectedPost?.images) && (
+            <img
+              className="h-32 mx-auto my-2 rounded"
+              src={
+                files.length > 0
```
Comment (pruthvirajmv) [inline src/components/PostInputModal.js]: `> 0` is redundant check

File: src/pages/Login.js [hunk 7]
```diff
@@ -23,10 +23,12 @@ export const Login = () => {
 
   const handleTestLogin = () => {
     dispatch(
-      login({ email: "adarshbalak@moments.com", password: "adarshbalak" })
+      login({
+        email: process.env.REACT_APP_TEST_EMAIL,
+        password: process.env.REACT_APP_TEST_PASSWORD,
```
Comment (pruthvirajmv) [inline src/pages/Login.js:28]: good to see this. thanks!

**Review by pruthvirajmv — COMMENTED** [review]
Hey Ravi,
Liked the way how you have abstracted the many things.

I have given minor changes comments on the code, please try to resolve
And regarding the UI please do add favicon, document title and also fix the profile logo 

Good work, keep it up!

**Comment by ravikumar0403** [comment]
> Hey Ravi, Liked the way how you have abstracted the many things.
> 
> I have given minor changes comments on the code, please try to resolve And regarding the UI please do add favicon, document title and also fix the profile logo
> 
> Good work, keep it up!

Thank you for the review. I'll make the necessary changes as mentioned.

**Pull request merged**